metadata-ingestion/src/datahub/ingestion/source/ldap.py

```diff
@@ -351,36 +382,42 @@ def build_corp_user_mce(
         last_name = attrs[self.config.user_attrs_map["lastName"]][0].decode()
         groups = parse_groups(attrs, self.config.user_attrs_map["memberOf"])
 
-        email = (
-            (attrs[self.config.user_attrs_map["email"]][0]).decode()
-            if self.config.user_attrs_map["email"] in attrs
-            else ldap_user
-        )
-        display_name = (
-            (attrs[self.config.user_attrs_map["displayName"]][0]).decode()
-            if self.config.user_attrs_map["displayName"] in attrs
-            else full_name
-        )
-        department_id = (
-            int(attrs[self.config.user_attrs_map["departmentId"]][0].decode())
-            if self.config.user_attrs_map["departmentId"] in attrs
-            else None
-        )
-        department_name = (
-            (attrs[self.config.user_attrs_map["departmentName"]][0]).decode()
-            if self.config.user_attrs_map["departmentName"] in attrs
-            else None
-        )
-        country_code = (
-            (attrs[self.config.user_attrs_map["countryCode"]][0]).decode()
-            if self.config.user_attrs_map["countryCode"] in attrs
-            else None
-        )
-        title = (
-            attrs[self.config.user_attrs_map["title"]][0].decode()
-            if self.config.user_attrs_map["title"] in attrs
-            else None
-        )
+        # We check two things:
+        # - attribute exists
+        # - attribute is not empty list
+        if attrs.get(self.config.user_attrs_map["email"]):
+            email = (attrs[self.config.user_attrs_map["email"]][0]).decode()
```

Comment:
can this logic be extracted into a method and use the method - say - `get_attr_or_none` everywhere. looks like a lot of repetition here..

```
def get_attr_or_none(attrs, key, default=None):
    return attrs[self.config.user_attrs_map[key]][0].decode() if attrs.get(self.config.user_attrs_map[key]) else default
```

also is it assumed that `attrs[self.config.user_attrs_map[key]]` is instance of  list ? Is is always true ? May help to add a check here to avoid errors.